Add back manual install content to mobile docs (#14343)
newrelic/docs-website: src/content/docs/mobile-monitoring/new-relic-mobile-ios/installation/spm-installation.mdx

---
title: Swift Package Manager manual installation
metaDescription: How to instrument to your iOS applications using Swift Package Manager.
---

> We recommend that you use our [guided install](https://onenr.io/0qwLv87gkj5) to set up iOS monitoring. However, if you need to manually install for any reason, follow the steps below to install the New Relic iOS agent with Swift Package Manager.

Reviewer: for any reason

## Install your iOS application [#installation]

As part of the installation process, New Relic automatically generates an [application token](/docs/mobile-apps/viewing-your-application-token). This is a 40-character hexadecimal string for authenticating each mobile app you monitor in New Relic.

To install and configure your iOS/tvOS application:

1. Go to **[one.newrelic.com](https://one.newrelic.com/all-capabilities)**.

> 2. If applicable: From the Mobile Apps list, select **Add a new app**.

Reviewer: Should Mobile Apps be bolded?

> 2. If applicable: From the Mobile Apps list, select **Add a new app**.

Author: ```suggestion
2. If applicable: From the **Mobile Apps** list, select **Add a new app**.
```

3. From the **Get Started** page, select **iOS** as the platform for mobile monitoring.
4. Type a name for your mobile app, then select **Continue**.

Continue with the steps to configure New Relic for mobile monitoring.

## Configure using Swift Package Manager

1. Select **File > Swift Packages > Add Package Dependency...**.

2. Add the Github URL of the Package file:

   ```
   https://github.com/newrelic/newrelic-ios-agent-spm
   ```

   <Callout variant="tip">
     If you receive an `artifact of binary target 'NewRelic' failed extraction: The operation couldn’t be completed. (TSCBasic.StringError error 1.)` error when extracting the package, please close Xcode, delete the Derrived Data folder, re-open Xcode, and try again.
   </Callout>

> 3. Select the NewRelic package product, select your target, and select Finish.

Reviewer: Should Finish be bolded?

> 3. Select the NewRelic package product, select your target, and select Finish.

Author: ```suggestion
3. Select the NewRelic package product, select your target, and select **Finish**.
```

   </Callout>

5. Add a build script to your target's **Build Phases**. Ensure the new build script is the very last build script. Then paste the following, replacing `APP_TOKEN` with your [application token](/docs/mobile-apps/viewing-your-application-token):
    ### **iOS agent 7.4.0 or higher**:
    ```
    ARTIFACT_DIR="${BUILD_DIR%Build/*}"
    SCRIPT=`/usr/bin/find "${SRCROOT}" "${ARTIFACT_DIR}" -type f -name run-symbol-tool | head -n 1`
    /bin/sh "${SCRIPT}" "APP_TOKEN"

    ```
    ### **iOS agent 7.3.8 or lower**:
    ```
    SCRIPT=`/usr/bin/find "${SRCROOT}" -name newrelic_postbuild.sh | head -n 1`

    if [ -z "${SCRIPT}"]; then
     ARTIFACT_DIR="${BUILD_DIR%Build/*}SourcePackages/artifacts"
     SCRIPT=`/usr/bin/find "${ARTIFACT_DIR}" -name newrelic_postbuild.sh | head -n 1`
    fi

    /bin/sh "${SCRIPT}" "APP_TOKEN"
    ```

   ### **OPTIONAL**:
    Add the following lines to your build script above the existing lines to skip symbol upload during debugging:
    ```
    if [ ${CONFIGURATION} = "Debug" ]; then
        echo "Skipping DSYM upload CONFIGURATION: ${CONFIGURATION}"
        exit 0
    fi
    ```

> 6. If there is a checkbox below Run script that says "Run script: Based on Dependency analysis" please make sure it is not checked.

Reviewer: Should Run script be formatted?

Reviewer: Also there should be a comma before please make sure

> 6. If there is a checkbox below Run script that says "Run script: Based on Dependency analysis" please make sure it is not checked.

Author: ```suggestion
6. If there is a checkbox that says "Run script: Based on Dependency analysis" please make sure it is unchecked.
```
